pkg/rescaffold/migrate.go

```diff
@@ -152,24 +152,27 @@ func kubebuilderCreate(store store.Store) error {
 	return nil
 }
 
-func kubebuilderGrafanaPlugin(store store.Store) error {
+func migrateGrafanaPlugin(store store.Store, src, des string) error {
 	// If the plugin is already in the plugin chain, we don't need call 'edit' method
 	// Because the plugin is already migrated in the previous step
 	plugins := store.Config().GetPluginChain()
-	if slices.Contains(plugins, grafanaPluginKey) {
-		return nil
-	}
-	// If the plugin is not in the plugin chain, we need to call 'edit' method to add the plugin
-	var grafanaPlugin struct{}
-	err := store.Config().DecodePluginConfig(grafanaPluginKey, grafanaPlugin)
-	// If the grafana plugin is not found, we don't need to migrate
-	if errors.As(err, &config.PluginKeyNotFoundError{}) {
-		return nil
-	}
-	if err != nil {
-		return fmt.Errorf("Failed to Decode Grafana Plugin: %s. %v", grafanaPluginKey, err)
+	if !slices.Contains(plugins, grafanaPluginKey) {
+		// If the plugin is not in the plugin chain, we need to call 'edit' method to add the plugin
+		var grafanaPlugin struct{}
+		err := store.Config().DecodePluginConfig(grafanaPluginKey, grafanaPlugin)
+		// If the grafana plugin is not found, we don't need to migrate
+		if errors.As(err, &config.PluginKeyNotFoundError{}) {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("Failed to Decode Grafana Plugin: %s. %v", grafanaPluginKey, err)
+		}
+		err = kubebuilderGrafanaEdit()
```

Comment:
Question:
Why do we need to run `kubebuilderGrafanaEdit` in such circumstance where:
> the **PluginChain** does not contain `grafana-plugin` but the `grafana-plugin` can loaded from the **config**?

Reply:
First, in my understanding, GetPluginChain() method will return the 'layout' field in the PROJECT file, but not plugins field. Correct me if wrong. @camilamacedo86 @varshaprasad96 
In this case, if we don't run `kubebuilder init --plugins grafana.kubebuilder.io/v1-alpha` , but run 'edit' when the project has been inited. The PluginChain will not contain the grafana plugin('layout' field), but it will appears in 'plugins' field. So we need to call 'edit' and 'copy file'.
But if we init the project using two plugins, (or only the grafana plugin itself). The layout field will contain 'grafana plugin', which could be retrieved by 'GetPluginChain()' methond. Since we will also init the **new project** with all plugins. We don't need to call 'edit', but need to 'copy file'. 
Does it make sense?

Reply:
If you find the Grafana plugin in the PROJECT file then you must to:
- When you run kb init use the plugin
- After copy the file which has custom metrics config option for it in the new directory where the project was scaffolded
- Then, run kb edit with the plugin so that if custom metrics were setup for it then, you will also re-generate them.

Reply:
I'm thinking if the logic can be something like:
1. If the origin PROJECT file does not contain the Grafana plugin, then exit
2. If the origin PROJECT file contains the Grafana plugin, which means the origin project uses it:
    2.1 Run `edit ...plugin ...grafana` to generate the basic grafana dir
    2.2 Copy the config.yaml to the new scaffold
    2.3 Run `edit ...plugin ...grafana` again to generate additional files.

I am imagining if we can get free from the 'complicated' logic...

Reply:
Cool! That's much clearer. I didn't understand 'edit' command before.

Reply:
+1 Agreed with @Kavinjsir's logic. If the plugin doesn't exist in the origin PROJECT file, then we need to run the edit command. We can exit immediately.